modules/core/src/main/java/net/ashald/envfile/providers/yaml/YamlEnvFileParser.java

```diff
@@ -4,14 +4,28 @@
 import net.ashald.envfile.providers.EnvFileParser;
 import org.yaml.snakeyaml.Yaml;
 
+import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.Set;
 
 @AllArgsConstructor
 public class YamlEnvFileParser implements EnvFileParser {
     private final Yaml yaml;
 
     @Override
     public Map<String, String> parse(String content) {
-        return yaml.load(content);
+        Map<String, Object> value = yaml.load(content);
+        Set<String> keys = value.keySet();
+        Map<String, String> result = new LinkedHashMap<>();
+        for (String key : keys) {
+            Object v = value.get(key);
+            if (v != null && v.getClass().equals(String.class)) {
+                result.put(key, (String) v);
+            } else {
+                String stringValue = String.valueOf(v);
```

Comment:
Just browsing the codebase and noticed your PR. If v can be null, wouldn't String.valueOf(null) throw an exception. Similarly, I think null is also meant to be a valid value in yaml files.